packages/common/src/urls/get-portal-api-url.ts

```diff
@@ -26,7 +25,11 @@ import { getPortalUrl } from "./get-portal-url";
  * @returns The portal API URL, defaults to `https://www.arcgis.com/sharing/rest`
  */
 export function getPortalApiUrl(
-  urlOrObject?: string | IPortal | IHubRequestOptions | IRequestOptions
+  urlOrObject?:
+    | string
+    | IHubRequestOptionsPortalSelf
```

Comment:
Similar, `IHubRequestOptionsPortalSelf` extends `IPortal` and only adds an _optional_ `user` property, so this should be non-breaking.

Reply:
Since IPortal has `[key: string]: any;`, unless there is a _strong_ need for `.user` to be typed, I think we can skip this and just check for `.user` (which you'd have to to anyhow), and then cast it via `as UserSession`